components/raftstore/src/store/peer_storage.rs

```diff
@@ -1127,12 +1127,14 @@ impl PeerStorage {
             0
         } else {
             fail_point!("raft_before_apply_snap");
+            let mut kv_wb = ready_ctx.take_kv_wb();
             self.apply_snapshot(
                 &mut ctx,
                 ready.snapshot(),
-                &ready_ctx.kv_wb(),
-                &ready_ctx.raft_wb(),
+                &mut kv_wb,
+                ready_ctx.raft_wb_mut(),
             )?;
+            ready_ctx.save_kv_wb(kv_wb);
```

Comment:
`Take/save` patterns are error prone and I'd rather not use them. Instead, at least at this call site, you can have a method on `ctx` that returns two mutable references, one to each writebatch pointer.

Reply:
Ok. I agree with you.